spec/wizards/schools/register_ect_wizard/national_insurance_number_step_spec.rb

```diff
@@ -0,0 +1,199 @@
+    ].each do |value|
+      it { is_expected.to allow_value(value).for(:national_insurance_number) }
+    end
+
+    %w[DA123456A
+       FA123456A
+       IA123456A
+       QA123456A
+       UA123456A].each do |value|
+      it { is_expected.not_to allow_value(value).for(:national_insurance_number) }
+    end
+  end
+
+  describe '#next_step' do
+    subject { wizard.current_step }
+
+    let(:school) { FactoryBot.create(:school, :state_funded) }
+    let(:wizard) { FactoryBot.build(:register_ect_wizard, current_step: :national_insurance_number, step_params:, school:) }
+
+    context 'when the ect is not found in TRS' do
+      before do
+        allow(::TRS::APIClient).to receive(:new).and_return(TRS::FakeAPIClient.new(raise_not_found: true))
+        subject.save!
+      end
+
+      it 'returns :not_found' do
+        expect(subject.next_step).to eq(:not_found)
+      end
+    end
+
+    context 'when the teacher completed induction' do
+      let(:teacher) { FactoryBot.create(:teacher, trn: '1234568') }
+
+      before do
+        fake_client = TRS::FakeAPIClient.new
+
+        allow(fake_client).to receive(:find_teacher).and_return(
+          TRS::Teacher.new(
+            'trn' => '1234568',
+            'firstName' => 'Kirk',
+            'lastName' => 'Van Houten',
+            'dateOfBirth' => '1977-02-03',
+            'induction' => {
+              'status' => 'Passed'
+            }
+          )
+        )
+        allow(::TRS::APIClient).to receive(:new).and_return(fake_client)
+        subject.save!
+      end
+
+      it 'returns :induction_completed' do
+        expect(subject.next_step).to eq(:induction_completed)
+      end
+    end
+
+    context 'when the teacher is exempt from induction' do
+      let(:teacher) { FactoryBot.create(:teacher, trn: '1234568') }
+
+      before do
+        fake_client = TRS::FakeAPIClient.new
+
+        allow(fake_client).to receive(:find_teacher).and_return(
+          TRS::Teacher.new(
+            'trn' => '1234568',
+            'firstName' => 'Kirk',
+            'lastName' => 'Van Houten',
+            'dateOfBirth' => '1977-02-03',
+            'induction' => {
+              'status' => 'Exempt'
+            }
+          )
+        )
+        allow(::TRS::APIClient).to receive(:new).and_return(fake_client)
+        subject.save!
+      end
+
+      it 'returns :induction_exempt' do
+        expect(subject.next_step).to eq(:induction_exempt)
+      end
+    end
+
+    context 'when the teacher failed induction' do
+      let(:teacher) { FactoryBot.create(:teacher, trn: '1234568') }
+
+      before do
+        fake_client = TRS::FakeAPIClient.new
+
+        allow(fake_client).to receive(:find_teacher).and_return(
+          TRS::Teacher.new(
+            'trn' => '1234568',
+            'firstName' => 'Kirk',
+            'lastName' => 'Van Houten',
+            'dateOfBirth' => '1977-02-03',
+            'induction' => {
+              'status' => 'Failed'
+            }
+          )
```

Comment:
This is repeated 3 times with the only difference being the induction status. 

It could be an idea to DRY it up by extracting it into a let block and then over-riding the induction status as required: 

```
let(:induction_status) { 'Failed' }

let(:teacher_data) do
  {
    'trn' => '1234568',
    'firstName' => 'Kirk',
    'lastName' => 'Van Houten',
    'dateOfBirth' => '1977-02-03',
    'induction' => {
      'status' => induction_status
    }
  }
end
```

Reply:
I just added a new commit with all your spec suggestions and a few more.

Reply:
Nice - specs look much neater now. Thanks @ltello! 🚢